File: Node/TypeORM_ORM/src/bll/services/employeeService.ts
import { Employees } from "dal/entities/Employees";
import { EmployeeRepository } from '../../dal/repositories/employeeRepository';

export class EmployeeService {
  private employeeRepository = new EmployeeRepository();

  async getAll() {
    return await this.employeeRepository.getAll();
  }

  async getById(id: number) {
    return await this.employeeRepository.getById(id);
  }

  async create(data: Partial<Employees>) {
    return await this.employeeRepository.create(data);
  }

  async update(id: number, data: Partial<Employees>) {
    return await this.employeeRepository.update(id, data);
  }

  async delete(id: number) {
    return await this.employeeRepository.delete(id);
  }
}

File: Go/GORM_ORM/bll/services/employeeService.go
package services

import (
	"GORM_ORM/dal/entities"
	"GORM_ORM/dal/repositories"
)

type EmployeeService interface {
	GetAll() ([]entities.Employee, error)
	GetByID(id int) (*entities.Employee, error)
	Create(employee *entities.Employee) error
	Update(employee *entities.Employee) error
	Delete(id int) error
}

type employeeService struct {
	repo repositories.EmployeeRepository
}

func NewEmployeeService(repo repositories.EmployeeRepository) EmployeeService {
	return &employeeService{repo: repo}
}

func (s *employeeService) GetAll() ([]entities.Employee, error) {
	return s.repo.GetAll()
}

func (s *employeeService) GetByID(id int) (*entities.Employee, error) {
	return s.repo.GetByID(id)
}

func (s *employeeService) Create(employee *entities.Employee) error {
	return s.repo.Create(employee)
}

func (s *employeeService) Update(employee *entities.Employee) error {
	return s.repo.Update(employee)
}

func (s *employeeService) Delete(id int) error {
	return s.repo.Delete(id)
}
